app/services/idm/fetch-user-roles-and-groups.service.js

```diff
@@ -16,11 +16,12 @@ const UserModel = require('../../models/idm/user.model.js')
  * This service looks up the user in the `idm` (identity management) schema and returns a combined array of all roles
  * (deduped in case the user is given the same role multiple times; for example, by being assigned a role directly, then
  * later added to a group which also includes that role). It also returns an array of groups that the user is a member
- * of.
+ * of, along with `userFound` to explicitly indicate whether or not the user id exists.
  *
  * @param {Number} userId The user id to get roles and groups for
  *
  * @returns {Object} result The resulting roles and groups
+ * @returns {UseModel[]} result.user Returns the UserModel representing the user, or `null` if the user is not found
```

Comment:
I found something! Think the JSDoc just needs tweaking as `result.user` is a single object rather than an array.

```suggestion
 * @returns {UseModel} result.user Returns the UserModel representing the user, or `null` if the user is not found
```

Reply:
Good catch! And I notice I missed the `r` out of `UserModel` too! 🙄